pkg/deployer/lib/controller.go

```diff
@@ -41,6 +42,10 @@ type Deployer interface {
 	ForceReconcile(ctx context.Context, di *lsv1alpha1.DeployItem, target *lsv1alpha1.Target) error
 	// Abort the deploy item progress.
 	Abort(ctx context.Context, di *lsv1alpha1.DeployItem, target *lsv1alpha1.Target) error
+	// AddExtensionHook adds an extension hook to this deployers hooks.
+	AddExtensionHook(hook extension.ReconcileExtensionHook)
```

Comment:
@schrodit Is the `AddExtensionHook` method needed? Or are the only ones who add hooks the deployers itself, which can directly access the hook list and have no need for an add method?